File: android/src/main/java/nl/bgma/installchecker/InstallCheckerPlugin.java
package nl.bgma.installchecker;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "InstallChecker")
public class InstallCheckerPlugin extends Plugin {

    private InstallChecker implementation;

    @Override
    public void load() {
        implementation = new InstallChecker(getContext());
    }

    @PluginMethod
    public void isInstalledFromStore(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("status", implementation.isInstalledFromStore());
        call.resolve(ret);
    }
}

File: ios/Sources/InstallCheckerPlugin/InstallCheckerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InstallCheckerPlugin)
public class InstallCheckerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "InstallCheckerPlugin"
    public let jsName = "InstallChecker"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "isInstalledFromStore", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = InstallChecker()

    @objc func isInstalledFromStore(_ call: CAPPluginCall) {
        call.resolve([
            "status": implementation.isInstalledFromStore()
        ])
    }
}
